libiop/bcs/bcs_verifier.tcc

```diff
@@ -57,19 +58,35 @@ void bcs_verifier<FieldT, hash_digest_type>::seal_interaction_registrations()
             }
         }
 
-        /* Absorb MT roots */
-        for (auto &kv : mapping)
+        /* Absorb MT roots into hashchain. */
+        const std::vector<hash_digest_type> MT_roots_for_round(
+            this->transcript_.MT_roots_.cbegin() + processed_MTs,
+            this->transcript_.MT_roots_.cbegin() + processed_MTs + num_oracles);
+        this->run_hashchain_for_round(round, MT_roots_for_round, this->transcript_.prover_messages_);
+
+        /* Validate all MT queries relative to the transcript. */
+        for (std::size_t i = 0; i < num_oracles; i++)
```

Comment:
This loop should be ran only once for every processed MT

Reply:
(So once per domain, not once per oracle) And the processed MT counter should be incrementing based off of this

Reply:
It's actually doing that, I just named the variables wrong